redfish-core/lib/hypervisor_system.hpp

```diff
@@ -167,94 +169,191 @@ inline bool extractHypervisorInterfaceData(
         {
             std::pair<boost::container::flat_set<IPv4AddressData>::iterator,
                       bool>
-                it = ipv4Config.insert(IPv4AddressData{});
+                v4Itr = ipv4Config.insert(IPv4AddressData{});
 
-            IPv4AddressData& ipv4Address = *it.first;
-            if (objpath.first == "/xyz/openbmc_project/network/hypervisor/" +
-                                     ethIfaceId + "/ipv4/addr0")
+            std::pair<boost::container::flat_set<IPv6AddressData>::iterator,
+                      bool>
+                v6Itr = ipv6Config.insert(IPv6AddressData{});
+
+            IPv4AddressData& ipv4Address = *v4Itr.first;
+            IPv6AddressData& ipv6Address = *v6Itr.first;
+
+            for (std::string protocol : {"ipv4", "ipv6"})
             {
-                idFound = true;
-                if (interface.first == "xyz.openbmc_project.Network.IP")
+                std::string ipAddrObj =
+                    "/xyz/openbmc_project/network/hypervisor/";
+                ipAddrObj.append(ethIfaceId);
```

Comment:
This isn't what upstream is doing

Reply:
@gtmills This was already mentioned - that we are based on 1020 downstream which is far behind the master code. We will have to refactor this as per the latest master when we do master rebase